### PR TITLE
Add support for removing a menu item

### LINKED ISSUE #33: Deleting Menu Items
I’m not well-versed enough in C/CGO to determine if hiding and nulling a menu or submenu item will trigger it for garbage collection.

If not, this is a feature request to be able to delete a menu/submenu item properly.  Use case is multiple dynamic submenus with a background goroutine populating/pruning items.

### PR DESCRIPTION
**Pull request by bdwyertech**
Resolves #33 and https://github.com/fyne-io/fyne/issues/3463

### CODE REVIEW

File: systray_darwin.m [hunk 1]
```diff
@@ -203,6 +203,14 @@ - (void) show_menu_item:(NSNumber*) menuId
   }
 }
 
+- (void) remove_menu_item:(NSNumber*) menuId
+{
+  NSMenuItem* menuItem = find_menu_item(menu, menuId);
```
Comment (andydotxyz) [inline systray_darwin.m:208]: This line, and the one 2 lines lower, seem to have bad indent

**Review by andydotxyz — CHANGES_REQUESTED** [review]
Nice addition, thanks.
I think some code is needed for unix as well?

**Comment by bdwyertech** [comment]
> Nice addition, thanks. I think some code is needed for unix as well?

I'll have to give the Unix code a further look.  Quick glance says pruning the button ID from a slice and triggering a refresh.

Also, I did notice another memory leak I cannot quite figure out... Seems to be related to the Icon setting code on Darwin specifically.  If I use a test harness to delete/create a new MenuItem every few seconds, after a few hours the memory usage multiplies exponentially.  pprof nor Datadog show anything.  If I remove SetIcon from the loop, the memory does not balloon nearly as badly.  My thought is something is not freeing up no longer used resources, probably something in the cast from CGO -> NSData -> NSImage -> NSMenuItem.image.

**Comment by andydotxyz** [comment]
Thanks, do post here if you need more assistance with the unix code.

With regards to icon memory leak, I think that also was reported in our usage in the toolkit at https://github.com/fyne-io/fyne/issues/3463, so you're not the only one to see it. The image data is obviously not being cleaned up in the C code.

**Comment by bdwyertech** [comment]
> Thanks, do post here if you need more assistance with the unix code.
> 
> With regards to icon memory leak, I think that also was reported in our usage in the toolkit at [fyne-io/fyne#3463](https://github.com/fyne-io/fyne/issues/3463), so you're not the only one to see it. The image data is obviously not being cleaned up in the C code.

Should all be resolved now.  Tested the tray on Kali, that dbus system is pretty interesting!

**Review by andydotxyz — APPROVED** [review]
Nice one thanks!

**Pull request merged**